Rename QueryIdCachingProxyHandler to Constants.java

RiyaJohn opened the pull request:
## Description

Fixes #480 

## Release notes

- (x) This is not user-visible or is docs only, and no release notes are required.

willmostly commented:
Thanks for the submission @RiyaJohn, this looks like it can be merged once cla-bot is happy. My only question is if we want to add a `utils` subpackage or put this class in an existing one. @oneonestar , @Chaho12 , @vishalya any opinion?

ebyhr commented on gateway-ha/src/main/java/io/trino/gateway/ha/util/Constants.java:
```diff
@@ -1,20 +1,7 @@
-package io.trino.gateway.ha.handler;
+package io.trino.gateway.ha.util;
+
+public class Constants {
```
Put `{` on a separate line and you could add `final`.

ebyhr replied:
The name is too general. Please rename to `TrinoHttpUtil` or something.

Chaho12 replied:
I like the part where it has Trino because this class is meant for Trino http spec.

ebyhr commented on gateway-ha/src/main/java/io/trino/gateway/ha/util/Constants.java:
```diff
@@ -1,20 +1,7 @@
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
```
Please don't remove license header.

ebyhr commented on gateway-ha/src/main/java/io/trino/gateway/ha/util/Constants.java:
```diff
@@ -26,6 +13,4 @@ public class QueryIdCachingProxyHandler
     public static final String AUTHORIZATION = "Authorization";
     public static final String USER_HEADER = "X-Trino-User";
-
-    private QueryIdCachingProxyHandler() {}
```
Please don't remove this constructor.

RiyaJohn replied:
you mean we have to retain QueryIdCachingProxyHandler class file with this constructor?

ebyhr replied:
No, I meant keeping a private constructor.

Chaho12 reviewed (approved):
LGTM if we rename the class name.
I think we should have it as a separate class because this is Trino http spec, not gateway.

mosabua reviewed (changes requested):
Having a generic util package and a generic Constants class is an antipattern .. it will just end up as a big undefined class with no focus and spread into the codebase all over as a dependency. Please keep it focussed like in the original issues from @ebyhr

RiyaJohn commented:
@ebyhr I think I have addressed the review comments, please take a look at it. And I need help with this cla-signing. I have an email configured and I can see it in the git config, so not sure what is missing

RiyaJohn commented:
@willmostly @Chaho12 @vishalya @ebyhr @mosabua Can someone help me here ^^^

mosabua commented:
@RiyaJohn .. you need to rebase your branch and squash the commits into one commit.

In addition you need to submit a signed CLA document and then wait for it to be verified and processed. Once that is done your github user name will show up in https://github.com/trinodb/cla/blob/master/contributors
and I can kick of another verification here.

ebyhr commented:
Superseded by #535
